### DISCUSSION TITLE
Bluetooth Connections - what does work and what doesn't

### DISCUSSION THREAD

**Discussion by stbuerger** [lkempf/casambi-bt-hass#91] [Show and tell]
Hi - with this I want to start a discussion topic on BT connection(s) working with the HA integration.
Personal experiences, observations, tips, tricks, and everything that cannot or should not be treated under "issues".

**Comment by stbuerger** [comment]
My current configuration and status:
Luminaires: several devices from OCCHIO (https://github.com/lkempf/casambi-bt/wiki/Working-Luminaires)
Casambi FW: Evolution/43.0
HAOS running in Proxmox VM
  Core 2024.11.0
  Supervisor 2024.11.2
  Operating System 13.2
  Frontend 20241106.0
Integration: [casambi-bt-hass](https://github.com/lkempf/casambi-bt-hass) currently (still) version 0.1.2
BT Connection: only via esphome-btproxy
BtProxy HW: M5 Atom Echo 
esphome-yaml: used from https://github.com/esphome/bluetooth-proxies/blob/main/m5stack/m5stack-atom-lite.yaml

Status:
Overall, the configuration works pretty well and is very reliable - when/if -  everything is running smoothly.
However, there are intermittent "interruptions" in service:
The network status entity intermittently transitions to a "not connected" state. Recovery to a "connected" state typically occurs within 30-60 seconds, but intervals between these transitions fluctuate between approximately 10 minutes and 1-3 hours. In some instances, a "re-connect" is delayed beyond 15 minutes, resulting in a comprehensive integration failure. In 80% of such occurrences, reloading the integration successfully resolves the issue. The frequency of these extended disconnections is highly variable, ranging from multiple instances within a 24-hour period to weeks of uninterrupted operation. On a few occasions over the past six months, the issue persisted, necessitating a restart of the BT-proxy hardware. To mitigate this erratic behavior, an automation has been implemented to automatically reload the integration following a sustained "not connected" state exceeding 10 minutes.

Based on these experiences and observations, I suspect the root cause lies in Bluetooth interference or issues with the BT proxy firmware. This suspicion is further supported by the significantly poorer performance observed when using different ESP32 hardware modules. Attempts to pinpoint the exact cause through extensive logging within the integration and BT proxy firmware have been temporarily abandoned.

**Comment by stbuerger** [comment]
Previous experience with various ESP32 BT proxy hardware:
"OK": Functions reliably for several hours, reconnects automatically, a "reload" of the integration helps with longer "not connected" states, a hardware restart is only rarely necessary.
"Poor": Disconnects every few minutes, a reload of the integration usually doesn't work either.
"Not Working": Doesn't work at all - integration doesn't start. 

OK:                  M5 Atom Echo (best so far), ESP32 D1 Mini (not quite as good)
Poor:               ESP32 DevKit(s), NodeMCU-32, Lolin-32 etc
Not Working: many different Shelly-32 including Shelly Blue Gateway

No experience with with BT adapters directly attached to / controlled by HAOS

**Comment by fschwarz86** [comment]
Thanks a ton for the details! I've flashed an M5 Atom Lite as a [Bluetooth Proxy](https://esphome.io/projects/?type=bluetooth) and it works like a charm!
